src/core_plugins/vega/public/vega_view/vega_base_view.js

```diff
@@ -32,6 +59,14 @@ export class VegaBaseView {
     this._initialized = true;
 
     try {
+      // Adapted from src/ui/public/filter_editor/filter_editor.js
+      if (!areIndexPatternsProvided(this._indexPatterns)) {
```

Comment:
This does properly not what you want. Where should `this._indexPatterns` come from in this place? This will always be the index patterns Angular service in your case? We need a way to specify the index pattern via the function because at the moment it will always create the filter on the default index pattern.

We should either accept an index pattern id to the `kibanaAddFilterHandler` method or alternative I think a index pattern itself, and we try to find that index pattern in the list of all index patterns.